repository/kafka/docs/v0.2/configuration.md

```diff
@@ -0,0 +1,70 @@
+# Configuration 
+
+### Resources
+
+By default, KUDO Kafka resource configuration is set to the minimum recommended values for production usage. 
+But users can and should tune the configurations based on the workload requirements of their Kafka cluster.  
+
+##### Tuning the resources for the Kafka Cluster
+
+```
+kubectl kudo install kafka --instance=my-kafka-name \
+            -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
+            -p ZOOKEEPER_PATH=/custom-path \
+            -p BROKER_CPUS=3000m \
+            -p BROKER_COUNT=5 \
+            -p BROKER_MEM=4096m \
+            -p DISK_SIZE=20Gi \
+            -p MIN_INSYNC_REPLICAS=3 \
+            -p NUM_IO_THREADS=10 \
+            -p NUM_NETWORK_THREADS=5 
+          
+```
+
+##### Storage
+
+By default, the Kafka operator will use the default storage class of the Kubernetes cluster. 
+
+To deploy Kafka using a different storage class, you can use the parameter `STORAGE_CLASS`
+
+```
+kubectl kudo install kafka --instance=my-kafka-name -p STORAGE_CLASS=<STORAGE_CLASS_NAME>
+```
+
+##### Deploying an ephemeral Kafka cluster without persistence
+
+```
+kubectl kudo install kafka --instance=my-kafka-name -p PERSISTENT_STORAGE=false
+```
+
+Having `PERSISTENT_STORAGE` value `false` means that any data or logs inside the brokers will be lost after a pod restart or rescheduling.
+Deploying without persistent storage isn't recommended for production usage. 
+
+##### Metrics
+
+By default, the Kafka cluster will have the JMX Exporter enabled. You can check more information around how KUDO Kafka exposes metrics in [monitoring](./monitoring.md).
+
+##### Zookeeper PATH
+
+By default, the Kafka cluster will try to use a zookeeper path same as specified in `--instance` flag.
+You can get the instance name by `kubectl get instances` 
```

Comment:
I don't understand most of this section...   This doesn't explain what the default is IMO.   It doesn't explain how the `--instance` flag is tied to the default zk.

Reply:
perhaps it doesn't make sense because although the topic is zk path... the main topic is kafka.    If the `--instance` flag is for the installation of zk.. that is unclear.   It is also unclear if that is the case how that is tied to the installation of kafka (unless we do the next step in this section... however this is stated as a default)

Reply:
so you don't understand `--instance` flag?

Reply:
`Zookeeper PATH` is a configuration of Kafka. And it has nothing to do with KUDO Zookeeper. It can be KUDO Zookeeper, DC/OS Zookeeper or living in bare metal. As long as the Kafka has access to the Zookeeper URI it will work.

Reply:
that's a bizarre question.... I completely understand what the `--instance` flag is...  however your reference to it doesn't make sense in the context of this documentation.   It seems you are trying to explain 3 options... 1) default, 2) path, 3) uri.     The distinction is lost on me as it is documented now.

Reply:
And you might add your last comment into the docs...   The fact that you can have a zookeeper any where should be in the docs.

Reply:
It's not bizarre,  you literally said `If the --instance flag is for the installation of zk.. that is unclear`. Which leads me to think that maybe I had to clarify what the `--instance` flag is for.

> you are trying to explain 3 options... 1) default, 2) path, 3) uri.

I don't see any reference to above URI here. 

> 1) default, 2) path

By default, the Kafka cluster will try to use a zookeeper path same as specified in `--instance` flag. 

Please feel free to open a PR with suggestions that you think will make these docs readability better.

Reply:
we need to make this readable and understandable before we merge... at this point it is hold up the merge.   Invitations to improve is always welcome but not a reason to allow something to continue that isn't understandable.

Reply:
How about we reword? 
```
##### Zookeeper PATH

Users must set the `ZOOKEEPER_PATH` to where they previously installed zookeeper in order to successfully deploy Kafka. 

To get the zookeeper instance name:
''' # this is meant to be triple backticks 
> kubectl get instances

zookeeper-abc123   1h
'''

Then install Kafka with: 
'''
kubectl kudo install kafka --instance=<my_kafka_name> \
  -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 \
  -p ZOOKEEPER_PATH=/zookeeper-abc123
'''
```
I wouldn't even mention the default. Would that even work? That would mean a user would deploy zookeeper under the same name as kafka? Can we make the default behavior assume `/zookeeper` instead? Then it's a little easier to understand and I would either add at the bottom or top of this section:
```
By default, Kafka will assume a user has already installed zookeeper with the name `/zookeeper`. Thus, a user can deploy Kafka more simply with:

'''
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zk-0.zk-hs:2181,zk-zk-1.zk-hs:2181,zk-zk-2.zk-hs:2181 
'''
```

Reply:
Oh wait, doesn't the `ZOOKEEPER_URI` also represent the zk instance name too? So here a user has previously deployed zookeeper as `zk` right? Hence `zk-zk-0, zk-zk-1,...`

Reply:
I have reworked this section: 

##### Zookeeper Configuration

KUDO Kafka requires a running ZooKeeper ensemble to perform its own internal accounting and persist cluster topology. 
You can install the KUDO Zookeeper or an alternate Zookeeper to use with KUDO Kafka. 

###### Configuring the Zookeeper connection:

You can configure KUDO Kafka to use Zookeeper using the KUDO Kafka parameter `ZOOKEEPER_URI`
Let's see this with an example:
```
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
```
In the above example KUDO Kafka cluster will connect to the Zookeeper cluster present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181`

The Kafka cluster topology is persisted in a Zookeeper node. Every node in Zookeeper's namespace is identified by a path.   

Two Kafka clusters can share the same Zookeeper but sharing the same path can lead to a corrupt state for both Kafka clusters. 
To avoid this to happen, KUDO Kafka persists its cluster topology in zk node with same name as KUDO Kafka instance.

Let's for example see what will the zk path in the above example where we just configured `ZOOKEEPER_URI`:
```
kubectl kudo install kafka --instance=my-kafka-cluster \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
```

In the above example KUDO Kafka will connect the Zookeeper present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181` and 
will create a Zookeeper node in path `/my-kafka-cluster`.  

###### Using a custom Zookeeper node path:

You can also decide to chose a custom Zookeeper node path, using the KUDO Kafka parameter `ZOOKEEPER_PATH`
Let's see this with an example:
```
kubectl kudo install kafka --instance=my-kafka-name \
  -p ZOOKEEPER_URI=zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181 \
  -p ZOOKEEPER_PATH=/custom-path
```
In the above example KUDO Kafka will connect the Zookeeper present in `zk-zookeeper-0.zk-hs:2181,zk-zookeeper-1.zk-hs:2181,zk-zookeeper-2.zk-hs:2181` and 
will create a Zookeeper node in path `/custom-path`.